sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowCreateTableExec.scala

```diff
@@ -0,0 +1,148 @@
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import scala.collection.mutable
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.util.escapeSingleQuotedString
+import org.apache.spark.sql.connector.catalog.{CatalogV2Util, Table, TableCatalog}
+import org.apache.spark.sql.connector.expressions.{BucketTransform, DaysTransform, HoursTransform, IdentityTransform, MonthsTransform, YearsTransform}
+import org.apache.spark.sql.errors.{QueryCompilationErrors, QueryExecutionErrors}
+import org.apache.spark.sql.execution.LeafExecNode
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * Physical plan node for show create table.
+ */
+case class ShowCreateTableExec(
+    output: Seq[Attribute],
+    table: Table) extends V2CommandExec with LeafExecNode {
+  override protected def run(): Seq[InternalRow] = {
+    val builder = StringBuilder.newBuilder
+    // it is used to generate Spark DDL for given table. include Hive Serde table
+    showCreateTable(table, builder)
+    Seq(InternalRow(UTF8String.fromString(builder.toString)))
+  }
+
+  private def showCreateTable(table: Table, builder: StringBuilder): Unit = {
+    builder ++= s"CREATE TABLE ${table.name()} "
+    val isHive = isHiveTable(table)
+    if (isHive) {
+      // need convert the SerdeInfo from table.properties
+      throw QueryCompilationErrors.showCreateTableNotSupportedForV2TablesError()
+    }
+    showTableDataColumns(table, builder)
+    showTableUsing(table, builder)
+    showTableOptions(table, builder)
+    showTableNonDataColumns(table, builder)
+    showTableComment(table, builder)
+    showTableLocation(table, builder)
+    showTableProperties(table, builder)
+  }
+
+  private def showTableDataColumns(table: Table, builder: StringBuilder): Unit = {
+    val columns = table.schema().fields.map(_.toDDL)
+    builder ++= concatByMultiLines(columns)
+  }
+
+  private def showTableUsing(table: Table, builder: StringBuilder): Unit = {
+    if (table.properties.containsKey(TableCatalog.PROP_PROVIDER)) {
+      builder ++=
+        s"USING ${escapeSingleQuotedString(table.properties().get(TableCatalog.PROP_PROVIDER))}\n"
+    }
+  }
+
+  private def showTableOptions(table: Table, builder: StringBuilder): Unit = {
+    import scala.collection.JavaConverters._
+    val dataSourceOptions = table.properties.asScala
+      .filterKeys(_.startsWith(TableCatalog.OPTION_PREFIX))
+    if (dataSourceOptions.nonEmpty) {
+      val props = dataSourceOptions.map { case (key, value) =>
+        s"'${escapeSingleQuotedString(key)}' = '${escapeSingleQuotedString(value)}'"
+      }
+
+      builder ++= "OPTIONS "
+      builder ++= concatByMultiLines(props)
+    }
+  }
+
+  private def showTableNonDataColumns(table: Table, builder: StringBuilder): Unit = {
+    if (!table.partitioning.isEmpty) {
+      val identityCols = new mutable.ArrayBuffer[String]
```

Comment:
`identityCols` looks weird, how about `transforms`?

Reply:
`transforms` is a better way to express meaning.